modules/network/src/main/java/org/apache/ignite/network/internal/netty/NettyClient.java

```diff
@@ -0,0 +1,126 @@
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network.internal.netty;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import io.netty.handler.stream.ChunkedWriteHandler;
+import java.net.InetSocketAddress;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import org.apache.ignite.network.message.MessageSerializationRegistry;
+import org.apache.ignite.network.message.NetworkMessage;
+
+/**
+ * Netty client channel wrapper.
+ */
+public class NettyClient {
+    /** Socket channel bootstrapper. */
+    private final Bootstrap bootstrap = new Bootstrap();
+
+    /** Socket channel handler event loop group. */
+    private final EventLoopGroup workerGroup = new NioEventLoopGroup();
+
+    /** Serialization registry. */
+    private final MessageSerializationRegistry serializationRegistry;
+
+    /** Destination host. */
+    private final String host;
+
+    /** Destination port. */
+    private final int port;
+
+    /** Incoming message listener. */
+    private final BiConsumer<InetSocketAddress, NetworkMessage> messageListener;
+
+    /** Future that resolves when client channel is opened. */
+    private final CompletableFuture<NettySender> clientFuture = new CompletableFuture<>();
+
+    /** Client socket channel. */
+    private Channel channel;
+
+    public NettyClient(String host, int port, MessageSerializationRegistry serializationRegistry, BiConsumer<InetSocketAddress, NetworkMessage> listener) {
+        this.host = host;
+        this.port = port;
+        this.serializationRegistry = serializationRegistry;
+        this.messageListener = listener;
+    }
+
+    /**
+     * Start client.
+     *
+     * @return Future that resolves when client channel is opened.
+     */
+    public CompletableFuture<NettySender> start() {
+        bootstrap.group(workerGroup)
+            .channel(NioSocketChannel.class)
+            /** See {@link NettyServer#start} for netty configuration details. */
+            .option(ChannelOption.SO_KEEPALIVE, true)
+            .handler(new ChannelInitializer<SocketChannel>() {
+                /** {@inheritDoc} */
+                @Override public void initChannel(SocketChannel ch)
+                    throws Exception {
+                    ch.pipeline().addLast(
+                        new InboundDecoder(serializationRegistry),
+                        new MessageHandler(messageListener),
+                        new ChunkedWriteHandler()
+                    );
+                }
+        });
+
+        ChannelFuture connectFuture = bootstrap.connect(host, port);
+
+        connectFuture.addListener(connect -> {
```

Comment:
this can be written a little bit shorter:
```
bootstrap.connect(host, port)
    .addListener((ChannelFutureListener)future -> {
        this.channel = future.channel();
        if (future.isSuccess())
            clientFuture.complete(new NettySender(channel, serializationRegistry));
        else
            clientFuture.completeExceptionally(future.cause());

        // Shutdown event loop group when channel is closed.
        channel.closeFuture().addListener(close -> {
            workerGroup.shutdownGracefully();
        });
    });
```